jsettlers.graphics/src/jsettlers/graphics/map/controls/original/panel/content/DistributionPanel.java

```diff
@@ -0,0 +1,286 @@
+                })),
+        ;
+        private final Icon icon;
+        private final ConfigurationPanel configurationPanel;
+
+        private Item(Icon icon, ConfigurationPanel configurationPanel)
+        {
+            this.icon = icon;
+            this.configurationPanel = configurationPanel;
+        }
+
+        public Icon getIcon()
+        {
+            return icon;
+        }
+
+        public ConfigurationPanel getConfigurationPanel()
+        {
+            return configurationPanel;
+        }
+    }
+
+    private static final float contentHeight_px = 216; // 360
+    private static final float contentWidth_px = 118; // 197
+
+    private static final float titleTop_px = 2;
+    private static final float titleTextHeight_px = 12;
+    private static final float iconSize_px = 18; // 30
+
+    private static final float marginTop_px = 23f;
+
+    private static final float titleTop = 1 - (titleTop_px / contentHeight_px);
+    private static final float titleTextHeight = titleTextHeight_px / contentHeight_px;
+
+    private static final float rowHeight = iconSize_px / contentHeight_px;
+    private static final float tileWidth = iconSize_px / contentWidth_px;
+    private static final float marginTop = 1 - (marginTop_px / contentHeight_px);
+    private static final float marginV = 3 / contentHeight_px;
+    private static final float marginH = 5 / contentWidth_px;
+    private static final int ROWS = 9;
+
+    private static final float rowHeight_px = 25f;
+    private static final float rowSpacingV_px = 8f;
+    private static final float rowTextHeight_px = 9f;
+    private static final float rowTextMarginBottom_px = 4f;
+    private static final float rowTextMarginLeft_px = 6f;
+//    private static final float rowText2MarginLeft_px = 8f;
+//    private static final float barWidth_px = 54f;
+
+    private static final float rowTextMarginLeft = rowTextMarginLeft_px / contentWidth_px;
+
+
+    private final UIPanel panel;
+    private Item item;
+    private final UIPanel selectionFrame;
+
+    public DistributionPanel()
+    {
+        selectionFrame = new UIPanel();
+        selectionFrame.setBackground(new OriginalImageLink(EImageLinkType.GUI, 3, 339, 0));
+
+        panel = new UIPanel();
+        panel.addChild(new Label(Labels.getString("controlpanel_distribution_title"), EFontSize.NORMAL), 0f, titleTop - titleTextHeight, 1f, titleTop);
+
+        Item[] items = Item.values();
+        int itemIdx = 0;
+        float top = marginTop;
+        for (int r = 0; r < ROWS; r++, top -= (rowHeight + marginV)) {
+            final Item item = items[itemIdx];
+            Icon icon = item.getIcon();
+            panel.addChild(
+                    items[itemIdx].getIcon(),
+                    marginH,
+                    top - rowHeight,
+                    marginH + tileWidth,
+                    top
+                    );
+            icon.setActionListener(new ExecutableAction() {
```

Comment:
Here you are setting an action listenerl. I know this works in this case, but we should at least add a comment that those static icons are used this way (and it would be good to make the DistributionPanel-Class singleton)